ConsoleApp1/Program.cs

```diff
@@ -1,39 +1,43 @@
 ﻿using System;
 using System.Text;
 using System.Net;
-using System.Threading;
 using System.Threading.Tasks;
 using NOnion;
+using Microsoft.FSharp.Core;
 
 namespace ConsoleApp1
 {
     class Program
     {
         static async Task Main(string[] args)
         {
-            var file = System.IO.File.OpenWrite("test.txt");
             var guard = await TorGuard.NewClientAsTask(IPEndPoint.Parse("199.184.246.250:443"));
-            var circuit = await TorCircuit.CreateFastAsTask(guard);
-            Console.WriteLine("Created circuit, Id: {0}", circuit.Id);
+            TorCircuit circuit = new (guard);
+            TorStream stream = new(circuit);
 
-            ManualResetEvent dataReceivedCompletion = new (false);
-
-            var stream = await TorStream.CreateDirectoryStreamAsTask(circuit);
-            stream.DataReceived.Subscribe(
-                data => file.Write(data, 0, data.Length),
-                _ => {
-                    Console.WriteLine("Received data size: {0}", file.Position);
-                    file.Flush();
-                    file.Close();
-                    dataReceivedCompletion.Set();
-                }
-            );
+            var circuitId = await circuit.CreateFastAsTask();
+            Console.WriteLine("Created circuit, Id: {0}", circuitId);
+            await stream.ConnectToDirectoryAsTask();
 
             var request = $"GET /tor/status-vote/current/consensus HTTP/1.0\r\nHost: 199.184.246.250\r\n\r\n";
             var requestBytes = Encoding.UTF8.GetBytes(request);
-            await stream.SendAsTask(requestBytes);
+            await stream.SendDataAsTask(requestBytes);
+            var response = await ReceiveAllAsString(stream);
+            System.IO.File.WriteAllText("test.txt", response);
+        }
+
+        static async Task<string> ReceiveAllAsString(TorStream stream)
+        {
+            var result = "";
+            var newMsg = await stream.ReceiveAsTask();
+
+            while (!FSharpOption<byte[]>.get_IsNone(newMsg))
```

Comment:
are you sure the property `newMsg.IsNone` doesn't exist?

Reply:
At least my VS auto-complete says it doesn't, seemed weird to me too.